machine-learning/solutions/seismic_salt_detection/trainer/inputs.py

```
@@ -44,16 +44,9 @@ def parse_csv(record):
     Returns:
         A dictionary with all column names and values for the record.
     """
-    distribution_defaults = [[0.0] for _ in range(constants.DISTRIBUTION_SIZE)]
-    weather_defaults = [[0.0] for _ in range(constants.WEATHER_SIZE)]
-    distribution_cols = ['distribution' +
-                         str(i) for i in range(constants.DISTRIBUTION_SIZE)]
-    weather_cols = ['weather' + str(i) for i in range(constants.WEATHER_SIZE)]
-    header_def = [[0.0], [''], [0], [0]] + \
-        distribution_defaults + weather_defaults
-    column_names = [TARGET_COLUMN, 'date', 'day',
-                    'hour'] + distribution_cols + weather_cols
-    columns = tf.decode_csv(record, record_defaults=header_def)
+    column_defaults = [[''], [0], [0]]
```

@wes-turner
wes-turner 6 days ago  Collaborator
Why does it make sense to have default 'depth' and 'salt'? If it does make sense, could you please explain in a comment?

  @alan-krumholz
alan-krumholz 5 days ago  Owner
record_defaults are a requirement in tf.decode_csv to deal with missing values when loading data. These are not used if there aren't missing values but tf needs these values (it also use them to infer the type [e.g. int, float, str])

  @wes-turner
wes-turner 21 hours ago  Collaborator
I think I still don't understand. I expect it to be better to either throw an exception or drop the record if data is missing. Why have you chosen to not mark all columns as required?

---

I see what you mean now. It is supposed to take empty Tensor objects to denote as required. Let me play with this to make sure I make it work as needed. Will update here once I have it working